### PR TITLE
Expand contributing documentation

### PR DESCRIPTION
**Pull request by raddevon**

### CODE REVIEW

File: docs/guides/contributing/code.rst [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+:edb-alt-title: Developing EdgeDB
 
 This section describes how to build EdgeDB locally, how to use its
 internal tools, and how to contribute to it.
 
+.. warning::
+
+    Pull requests without thorough testing are not likely to be merged. If
+    you're not sure if yours is well-tested enough, go ahead and submit. We can
+    help guide you to the finish line.
```
Comment (1st1) [inline docs/guides/contributing/code.rst:16]: This sounds a tad harsh to me. How about:

```suggestion
    Code-changing pull requests without adding new tests might take
    longer time to be reviewed and merged.
```

**Comment by raddevon** [comment]
Forgot to tag you for a review on this, @jaclarke. If you have a chance, see if you notice any glaring omissions or inaccuracies.

File: docs/guides/contributing/documentation.rst [hunk 2]
```diff
@@ -0,0 +1,1158 @@
...
+
+**Rendered**
+
+See :ref:`the "Globals" documentation page <ref_datamodel_globals>` for a
+full-page example of ``.. versionadded:: 2.0``.
+
+Changed in Version
+------------------
+
+Use the ``versionchanged`` directive to mark content related to a change in to
```
Comment (vpetrovykh) [inline docs/guides/contributing/documentation.rst:1032]:
```suggestion
Use the ``versionchanged`` directive to mark content related to a change in
```

File: docs/guides/contributing/index.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+============
+Contributing
+============
+
+:edb-alt-title: Contributing to EdgeDB
+
+EdgeDB is an open-source project, and we welcome contributions from our
+community. You can contribute by writing code or by helping us improve our
+documentation .
```
Comment (vpetrovykh) [inline docs/guides/contributing/index.rst:11]:
```suggestion
documentation.
```

**Review by vpetrovykh — CHANGES_REQUESTED** [review]
Aside from a couple of typos LGTM.

**Review by vpetrovykh — APPROVED** [review]

File: docs/guides/contributing/documentation.rst [hunk 4]
```diff
@@ -0,0 +1,1158 @@
...
+and `follow the installation instructions
+<https://github.com/edgedb/website#installation>`_. Then run ``yarn dev`` to
+start a development server which also triggers a build of all the
+documentation.
+
+.. note::
+
+    Although the development server runs a watch task, it does not update
+    documentation as part of this. When you make changes to documentation, you
+    will need to interrupt and restart the development server.
```
Comment (jaclarke) [inline docs/guides/contributing/documentation.rst:133]: This behaviour is a bug. `yarn dev` should watch for docs changes and re-build automatically without needing to restart (with the small caveat that we can't communicate this update to nextjs hot reload, so you do have to manually refresh the page in the browser).

**Pull request merged**